src/wp-includes/functions.php

```diff
@@ -4947,14 +4947,31 @@ function _wp_array_get( $input_array, $path, $default_value = null ) {
 	}
 
 	foreach ( $path as $path_element ) {
-		if (
-			! is_array( $input_array ) ||
-			( ! is_string( $path_element ) && ! is_integer( $path_element ) && ! is_null( $path_element ) ) ||
-			! array_key_exists( $path_element, $input_array )
-		) {
+		if ( ! is_array( $input_array ) ) {
```

Comment:
Does it seem weird to you that this is inside the loop? `$input_array` is documented to be an array. Is it ever sent something else that just returns the same value at the end of this function?

Reply:
Agreed, this check can and should be before the loop. But it's also overly defensive, so I'd probably just remove it altogether.

Reply:
Okay I realize now that the loop modifies `$input_array` in place. Now it makes more sense.

Reply:
You got it. It needs to be in the loop as the loop goes deeper and deeper into the array and sub-items may not always be arrays.